### PR TITLE
Add PKCS#11 documentation

### PR DESCRIPTION
**Pull request by neusdan**
#694

### CODE REVIEW

File: doc/manual/pkcs11.rst [hunk 1]
```diff
@@ -0,0 +1,1244 @@
...
+
+   .. cpp:function:: PKCS11_ECDSA_KeyPair PKCS11::generate_ecdsa_keypair(Session& session, const EC_PublicKeyGenerationProperties& pub_props, const EC_PrivateKeyGenerationProperties& priv_props)
+
+----------
+
+Code example:
+
+   .. code-block:: cpp
+
+      Botan::PKCS11::secure_string pin = { '1', '2', '3', '4', '5', '6' };
```
Comment (randombit) [inline doc/manual/pkcs11.rst]: This is not related to the documentation, but I am seeing a need for a `std::string` to `secure_string` conversion function here. ISTM like nearly every application using the PKCS11 interface will end up writing such a function.

**Comment by mouse07410** [comment]
@neusdan, great! **Thank you!**

A question: I would like to use keys on a PKCS#11 token with TLS (that Botan also provides). How can I do that? A small example would be *very much* appreciated!

**Comment by neusdan** [comment]
AFAIK the TLS API currently does not support using other providers.

**Comment by mouse07410** [comment]
:-(

It would be great if all of the PK operations (where it makes sense) could use keys on a PKCS#11 token.

And it would be great if tls client and server could use it. Like OpenSSL does via pkcs11 engine.

**Comment by cordney** [comment]
Discussing on a closed PR seems wrong, please use #730.

**Pull request merged**